python/ray/_private/accelerator.py

```diff
@@ -194,31 +194,45 @@ def _autodetect_tpu_version() -> Optional[str]:
     """
 
     def accelerator_type_to_version(accelerator_type: str) -> str:
-        assert_tpu_accelerator_type(accelerator_type)
-        return "TPU-" + str(accelerator_type.split("-")[0]).upper()
+        if valid_tpu_accelerator_type(accelerator_type):
+            return "TPU-" + str(accelerator_type.split("-")[0]).upper()
+        else:
+            return None
 
+    detected_tpu_version = None
     # GKE-based check
     accelerator_type = os.getenv(
         ray_constants.RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR, None
     )
     if accelerator_type is not None:
-        return accelerator_type_to_version(accelerator_type)
-
-    # GCE-based VM check
-    try:
-        accelerator_type_request = requests.get(
-            ray_constants.RAY_GCE_TPU_ACCELERATOR_ENDPOINT,
-            headers=ray_constants.RAY_GCE_TPU_HEADERS,
-        )
-        if accelerator_type_request.status_code == 200:
-            return accelerator_type_to_version(accelerator_type_request.text)
-    except requests.RequestException as e:
-        logging.info("Unable to poll TPU GCE metadata: %s", e)
-
-    return None
-
-
-def assert_tpu_accelerator_type(accelerator_type: str):
+        detected_tpu_version = accelerator_type_to_version(accelerator_type)
+    else:
```

Comment:
Just want to confirm that by introducing this `else` block, polling the GCE endpoint is no longer on the critical path of `ray.init()`?  In other words, the GCE endpoint is only polled if `RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR` is set by the user.

Reply:
No, if the env var RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR is not set, then we would poll the GCE endpoint. If it is set, we don't poll the endpoint.

The difference is now if the poll fails, then nothing happens - we just log that we couldn't detect the TPU type (vs. raising an error like we did before)

Reply:
Actually, I misread the code, it seems we only poll the GCE endpoint if the env var is *not* set by the user.  So in this case, `requests.get()` is still on the path of the default `ray.init()` call.  Would it make sense to gate this behind another environment variable?

Reply:
Ah sorry just saw your response

Reply:
Want to get your and @rkooo567's thoughts on gating the `requests.get()` call, I'm not sure how risky it is to leave it on the default path (like if it could cause hangs or cause `ray.init()` to become slow)

Reply:
One way we could gate the requests.get is by only running the function when we're sure we have a particular accelerator type, e.g. [here](https://github.com/ray-project/ray/blob/baa861dc5a7967dd903d1b64e88277e3e76d6047/python/ray/_private/accelerator.py#L109-L117) by replacing [this](https://github.com/ray-project/ray/blob/baa861dc5a7967dd903d1b64e88277e3e76d6047/python/ray/_private/accelerator.py#L49) with an executable function, but honestly it's not very elegant.

I don't think requests.get should have drastic impact, but then again it already has 😛 happy to hear further thoughts though.

Reply:
If the logging goes straight to a file and not to the terminal output though, I think that could be acceptable.  But we wouldn't want it to be too hidden, for those people who actually need to see those logs

Reply:
Maybe for now we could just go with the approach for the autodetect function to trigger only if we detected an accelerator already?

Reply:
That sounds good to me!

Reply:
Done!

Reply:
So, the requirement here is; 
1. We should avoid using request.get which is blocking
2. We should not print TPU not detected logs to users when they don't use TPUs.

Haven't followed it closely here. Is this PR satisfying these 2 requirements?

Reply:
1. Yes on the default path.  But if TPUs are detected, we do use `request.get()` to get the type of TPUs.
2. Yes

Reply:
That sounds perfect!